packages/react-tinacms-github/src/github-client/GithubFile.ts

```diff
@@ -0,0 +1,83 @@
+/**
+
+Copyright 2019 Forestry.io Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+import { useMemo } from 'react'
+import { useCMS } from '@tinacms/react-core'
+
+type parseFn = (content: string) => any
+type serializeFn = (data: any) => string
+
+export class GithubFile {
+  private sha: string | null = null
+
+  constructor(
+    private cms: any,
+    private path: string,
+    private parse?: parseFn,
+    private serialize?: serializeFn
+  ) {}
+
+  fetchFile = async () => {
+    const res = await this.cms.api.github.fetchFile(this.path)
+    this.sha = res.sha
+    return this.parse ? this.parse(res.content) : res.content
+  }
+
+  commit = async (
+    data: any,
+    message: string = 'Update from TinaCMS',
+    retryOnConflict = true
```

Comment:
When would you not want to retryOnConflict?

Reply:
If there are irreconcilable conflicts with upstream changes it would just retry forever. We only want to retry when the conflict is due to a stale SHA, so retrying exactly once seems suitable.

Reply:
To put it a different way: a conflict that can't be resolved with a single retry will not be resolved by this system.

I suppose if we directly called commit from the registered API on line 62 instead of recursively calling this method again, we wouldn't need this guard parameter